grammar.js

```diff
@@ -1339,6 +1448,146 @@ module.exports = grammar({
+      ),
+    ),
+
+    alter_role: $ => seq(
+      $.keyword_alter,
+      choice(
+        $.keyword_role,
+        $.keyword_group,
+        $.keyword_user,
+      ),
+      choice($.identifier, $.keyword_all),
+      choice(
+        $.rename_object,
+        seq(optional($.keyword_with),repeat($._role_options)),
+        seq(
+          optional(seq($.keyword_in, $.keyword_database, $.identifier)),
+          choice(
+            seq(
+              $.keyword_set,
+              field("option", $.identifier),
+              choice(
+                seq($.keyword_from, $.keyword_current),
+                $.set_configuration,
+              ),
+            ),
+            seq(
+              $.keyword_reset,
+              choice(
+                $.keyword_all,
+                field("option", $.identifier),
+              )),
+          ),
+        )
+      ),
+    ),
+
+    set_configuration: $ => seq(
+      choice($.keyword_to, "="),
+      choice(
+        field("parameter", $.identifier),
+        $.literal,
+        $.keyword_default
+      )
+    ),
+
+    alter_index: $ => seq(
+      $.keyword_alter,
+      $.keyword_index,
+      optional($._if_exists),
+      $.identifier,
+      choice(
+        $.rename_object,
+        seq(
+          $.keyword_alter,
+          optional($.keyword_column),
+          alias($._natural_number, $.literal),
+          $.keyword_set,
+          $.keyword_statistics,
+          alias($._natural_number, $.literal),
+        ),
+        seq($.keyword_reset, paren_list($.identifier)),
+        seq(
+          $.keyword_set,
+          choice(
+            seq($.keyword_tablespace, $.identifier),
+            paren_list(seq($.identifier, '=', field("value", $.literal)))
+          ),
+        ),
+      ),
+    ),
+
+    alter_sequence: $ => seq(
+      $.keyword_alter,
+      $.keyword_sequence,
+      optional($._if_exists),
+      $.object_reference,
+      choice(
+        seq(
+          optional(seq(
+            $.keyword_start,
+            optional($.keyword_with),
+            field("start", alias($._integer, $.literal)),
+            optional(seq($.keyword_cache, field("cache", alias($._integer, $.literal)))),
+            optional(seq(optional($.keyword_no), $.keyword_cycle))
+          )),
+          optional(seq($.keyword_owned, $.keyword_by, choice($.keyword_none, $.object_reference))),
+          optional(seq(
+            $.keyword_start,
+            optional($.keyword_with),
+            field("start", alias($._integer, $.literal)),
+            optional(seq($.keyword_cache, field("cache", alias($._integer, $.literal)))),
+            optional(seq(optional($.keyword_no), $.keyword_cycle))
+          )),
+          optional(seq(
+            $.keyword_restart,
+            optional($.keyword_with),
+            field("restart", alias($._integer, $.literal)),
+          )),
+          optional(seq($.keyword_owned, $.keyword_by, choice($.keyword_none, $.object_reference))),
+        ),
```

Comment:
this is suspect -- there are a lot of options in the docs but the `start` sequence is repeated, and it's not possible to `alter sequence x no cycle` etc because those options are locked into the `start` sequence. I think something like this is the best result, a seq of optional seqs in the [documentation order](https://www.postgresql.org/docs/current/sql-altersequence.html):

```
seq(
  optional(seq($.keyword_as), <type>),
  optional(seq($.keyword_increment, optional($.keyword_by), $.literal)),
  optional(seq($.keyword_minvalue, choice($.literal, 'no minvalue')),
  optional(seq($.keyword_maxvalue, choice($.literal, 'no maxvalue')),
  optional(seq($.keyword_start, optional($.keyword_with), $.literal)),
  etc
)
```

Reply:
Oh, there are some copy and paste mishaps. I have fixed that, added the missing variants and I accept my questionable formatting :-)